openhtf/io/output/output_to_file.py

```diff
@@ -0,0 +1,32 @@
+"""Output to file.
+
+Output modules inherit from OutputToFile and overload methods
+based on output file requirements.
+
+"""
+
+import os
+import tempfile
+
+
+class OutputToFile(object):
+  """Class for saving to file."""
+
+  def Save(self):
+    """Save to file or file object."""
+    self.filename = None
+    if isinstance(self.filename_pattern, basestring):
+      if '{' in self.filename_pattern:
+        self.filename = self.filename_pattern.format(**self.test_record_dict)
+      else:
+        self.filename = self.filename_pattern % self.test_record_dict
+      self.SaveFile()
+    else:
+      self.filename_pattern.write(self.data)
+    return self.filename
+
+  def SaveFile(self):
+    with tempfile.NamedTemporaryFile(delete=False) as temp:
```

Comment:
See other comments, I think @amyxchen has a `util` function for you that will do this.